### ISSUE TITLE
Can the for-loop over the Connection object be stopped ?

### ISSUE THREAD

**Issue by MoaMoaK** [notti/fnfqueue#2]
__TLDR:__  
All the examples provided are using infinite loops to catch the packets but is there a way to properly stop this loop (without interrupting brutally with CTRL-C or kill or ...) ?

__Detailed situation:__  
I have a thread with a simple for-loop executing a function for each packet found in NFQUEUE, basically something like:
```python
for packet in conn:
    packet.payload = my_func(packet.payload)
    packet.mangle()
```
While this thread is running endlessly, I execute some commands in the main thread (whose network packets are caught by nfqueue and end-up in the above code).

But at the end, when I'm finished executing commands in the main thread, I want to finish the thread and exit properly the program. Which I can't do because the `Connection.__next__()` method is blocked trying to fetch next packet (or it is blocked in libnfqueue somewhere, I don't know). And I don't want to be forced to send a packet just to unblock the execution and trigger a `break` in the for-loop.

I even tried to use [killable threads](http://tomerfiliba.com/recipes/Thread2/) but they don't seems to work (from what I see it is because the execution is waiting in C code and not python bytecode, but again I'm not sure as I'm not familiar with this).

So what is the recommended way to properly stop the for-loop if there are not more packets coming ? (If there is one)

**Comment by notti** [comment]
At the moment this is not implemented; I just had an idea how to do that, but discovered that error handling is somehow completely broken.
I'll fix that and then try my idea - might take a couple days.

**Comment by notti** [comment]
@MoaMoaK Can you try out the current master, which adds support for stopping the loop by calling `Connection.close()`?
I'll draw a new release if this works.

**Comment by MoaMoaK** [comment]
@notti Great this is working as expected. Thanks for the reactivity :)
Here is a the test I did : https://gist.github.com/MoaMoaK/a335de592085d7b5dcc722d39869d455

**Comment by notti** [comment]
I made a new release just now, that includes this fix (version 1.1.0 on pypi).
Closing this, as it is solved now.
